src/mgclient.c

```diff
@@ -539,24 +499,29 @@ static int get_hostname_and_ip(const struct sockaddr *peer_addr, char *hostname,
         return MG_ERROR_NETWORK_FAILURE;
       }
       break;
-
     default:
       // Should not happen with addresses returned from getaddrinfo.
       abort();
   }
-
-  int status = getnameinfo(peer_addr, sizeof(struct sockaddr), hostname,
-                           NI_MAXHOST, NULL, 0, 0);
-  if (status != 0) {
-    mg_session_set_error(session, "failed to get server name: %s",
-                         gai_strerror(status));
-    return MG_ERROR_NETWORK_FAILURE;
+  // Populate the hostname.
+  // Useful read https://stackoverflow.com/questions/12274028.
+  int nameinfo_status = getnameinfo(peer_addr, sizeof(struct sockaddr),
+                                    hostname, NI_MAXHOST, NULL, 0, 0);
+  if (nameinfo_status != 0) {
```

Comment:
What if this fails because of some other reason? You would also pass it as a valid get_hostname_and_ip

Reply:
The hostname defined here is used in the SSL related user-defined trust callback. The hostname will be passed to the user code once the callback is called. In case of any failure, the user will get `unknown` and has to decide what to do (user code has to deal with that situation). Previous code is problematic because the whole client doesn't work because that `getnameinfo` failed, even if SSL is NOT used (it failed in a trivial case when `peer_addr` was constructed from `localhost` on Windows, a common case). I was thinking of adding a warning, but it doesn't make sense because in the trivial case user will get a useless alert (he doesn't deal with the callback at all). I see two options:
1) I have to figure out what is going on with these two calls, `getaddrinfo` and `getnameinfo`. That's all connected and quite complex because it's platform dependent.
2) Explain to the user what is going on in case he gets `unknown` + a guide on how to handle it.

I would prefer option two at this point.

Reply:
The third option is to combine. `get_hostname_and_ip` can be placed under `MG_SSLMODE_REQUIRE` + in the failure case, a warning can be logged + the hostname can be something equivalent to `unknown`?

Reply:
Hmm, I agree that it doesn't make sense to LOG it to the user as he can't do much about it if I understood it correctly. But adding DLOG would be great to have!